sdk/storage/azure-storage-blob/azure/storage/blob/_quick_query_helper.py

```diff
@@ -27,25 +31,57 @@ class QuickQueryReader(object):  # pylint: disable=too-many-instance-attributes
 
     def __init__(
         self,
-        client=None,
         name=None,
         container=None,
-        progress_callback=None,
+        errors='strict',
         encoding=None,
-        **kwargs
+        headers=None,
+        response=None
     ):
         self.name = name
         self.container = container
-        self.response_headers = None
-        self.total_bytes = None
+        self.response_headers = headers
+        self.size = None
         self.bytes_processed = 0
-        self._progress_callback = progress_callback
-        self._client = client
+        self._errors = errors
         self._request_options = kwargs
         self._encoding = encoding
+        self._parsed_results = DataFileReader(QuickQueryStreamer(response), DatumReader())
+        self._first_result = self._process_record(next(self._parsed_results))
 
     def __len__(self):
-        return self.total_bytes
+        return self.size
+
+    def _process_record(self, result):
+        self.size = result.get('totalBytes')
+        self.bytes_processed = result.get('bytesScanned')
+        if result.get('data'):
+            return result.get('data')
+        if result.get('fatal') is not None and self._errors != 'ignore':
+            error = BlobQueryError(
+                error=result['name'],
+                is_fatal=result['fatal'],
+                description=result['description'],
+                position=result['position']
+            )
+            if result['fatal'] or self._errors == 'strict':
+                raise error
+            if self._errors == 'warning':
+                _LOGGER.warning(str(error))
+            else:
+                self._errors(error)
+        return None
+
+    def _iter_records(self):
```

Comment:
we should rename _iter_records, because later probably we want to iterate the real records(json or csv records) as we discussed offline.
Here, it's iterating avro parsed data according to service provided schema.

Reply:
I've already added the record iterator in the `def records()` function :)
It separates the data based on the specified (or default) record delimiter.

Reply:
if this name is iter_records, the expected item in the record list(or generator) is a record, while records() is delimiting the "record" in to real records, so I think we should rename iter_records to separate this "record" with the real record

Reply:
This is an internal function - so it's name isn't so important.
But I will rename it to `def _iter_stream` to make it more clear.
The records iterator returned by `def records` does actually separate out the individual records from the stream.